tests/set/set5.c

```diff
@@ -0,0 +1,33 @@
+#include "assert.h"
+#include <stdlib.h>
+#include <adt/set.h>
+
+int cmp_int(const void *a_, const void *b_) {
+	int a = *(const int *)a_, b = *(const int *)b_;
+	if (a > b)      return 1;
+	else if (a < b) return -1;
+	else            return 0;
+}
+
+int *next_int(void) {
+	static int n = 0;
+	int *p = malloc(sizeof *p);
+	assert(p);
```

Comment:
`abort()` for malloc - because it's not what the test is testing for, and because the test still ought to be well-behaved C even if -DNDEBUG is specified.